Azure/sonic-buildimage
# [docker-orchagent]: Add NDP Proxy Daemon

Signed-off-by: Lawrence Lee <lawlee@microsoft.com>


Add NDP Proxy Daemon to the `docker-orchagent` image, and manage it with `supervisord`.

**- Why I did it**
Native proxy NDP in Linux requires explicit neighbor entries in the kernel neighbor table, i.e. it's not possible to configure a subnet for proxy NDP. NDP Proxy Daemon (`ndppd`) bypasses the kernel neighbor table entirely and allows for subnets when configuring proxy NDP.

**- How I did it**
* Install `ndppd` during image build, and copy config files to image
* Configure proxy settings based on config DB at container start
* Pipe `ndppd` output to `logger` inside container to log output in syslog

**- How to verify it**
1. Copy and load `docker-orchagent` image to a device that supports `proxy_arp`.
2. Enable `proxy_arp` for a VLAN interface in config_db.
3. Stop and delete the existing `swss` container.
4. `config reload`
5. Select an unused IPv6 address in the same subnet as one of the VLAN interfaces on the device from step 1.
6. On a neighbor connected to the device from step 1, choose an interface connected to the VLAN used in step 5, and add the IPv6 address from step 5 to this interface.
7. From the neighbor, attempt to ping a new, unused IPv6 address in the same subnet as the VLAN interface from step 5 (the ping should fail).
8. Check the output of `ip -6 neigh`. The IPv6 address pinged in step 7 should be `REACHABLE`.

**- Which release branch to backport (provide reason below if selected)**




**- Description for the changelog**

Add support for proxy NDP

**- A picture of a cute animal (not mandatory but encouraged)**
